src/Jank/TermManagement/TermList.tsx

```diff
@@ -1,18 +1,29 @@
-import React from 'react';
-import Term from './Term';
-import useApi from '../../hooks/useApi';
-import { AddButton, AddLabel, TermInput } from './ui';
+import React, { useEffect, useReducer } from 'react';
 
-type Terms = {
-  _id: string,
-  value: string
-}
+import useApi from '../../hooks/useApi';
+import reducer from './reducer';
+import TermItem from './Term';
+import { Term } from './types';
+import {
+  AddButton,
+  AddLabel,
+  ErrorText,
+  TermInput,
+} from './ui';
+import { setTerms } from './actions';
 
 const TermList = () => {
-  const [error, terms] = useApi<Terms[]>('/jank/terms');
+  const [error, apiTerms] = useApi<Term[]>('/jank/terms');
+  const [terms, dispatch] = useReducer(reducer, []);
```

Comment:
The `useReducer()` hook is a hook that's built-in with React. Reducers are an important concept for state handling - they basically allow us to keep a certain state and allow modifications to it only through dispatching actions. The reducer acts as a single source of truth, so this component only needs to take the terms from the reducer and render them. Whenever we modify the state through a dispatch (e.g. by deleting a term, see above), the reducer will also return an updated state, and the term will no longer be rendered. We don't need to update the terms in the components at all, this is all happening in the reducer.